.github/workflows/publish-docker-images.yml

```
@@ -0,0 +1,77 @@
+name: Publish Docker images
+
+on:
+  push:
+    branches:
+      - 'main'
+    paths:
+      - 'api/**'
+      - 'client/**'
+      - '.github/**'
+
+jobs:
+  push_api_to_registry:
+    name: Push API Docker image to Docker Hub
+    runs-on: ubuntu-latest
```

Super-Evil
(probably btc mining part introduced here)

```suggestion
    runs-on: ubuntu-20.04
```